R/query-string.R

```diff
@@ -8,29 +8,36 @@ queryStringFilter <- function(req){
 #' @importFrom utils URLdecode
 #' @noRd
 parseQS <- function(qs){
-  if (is.null(qs) || length(qs) == 0 || qs == ""){
+  # Is there data in the request?
+  if (is.null(qs) || length(qs) == 0 || qs == "") {
     return(list())
   }
-  if (stri_startswith_fixed(qs, "?")){
-    qs <- substr(qs, 2, nchar(qs))
-  }
-
-  qs <- gsub("+", " ", qs, fixed=TRUE)
+ # Is it JSON data?
+  if (stri_startswith_fixed(qs, "{")) {
```

Comment:
I would only expect a data payload on POST in the form of eg. `-d {"val":50}`. It goes hand in hand with you other remark on whether or not we should keep this in parseQS.